commitizen/cli.py

```diff
@@ -237,6 +238,18 @@
     },
 }
 
+original_excepthook = sys.excepthook
+
+
+def commitizen_excepthook(type, value, tracekback):
```

Comment:
neat! cool!
What do you think if the exception also handles the error message?

Instead of:
```
out.error(f"'{self.config.settings['name']}' rule does not support bump")
raise NoPatternMapError()
```

```
raise NoPatternMapError(f"'{self.config.settings['name']}' rule does not support bump")
```

and this hook takes care of `out.error` if message exists.

Reply:
Yes, that's what I'm currently working on! I'm thinking of adding a default message while supporting a customized message. Also, some of these exceptions only show as a warning.

BTW, I notice `--debug` is deprecated after 2.0. May I know the reason of it? I'm thinking of keeping the `--debug` flag and handle it through expcetionhook so that users can get extra information (like traceback) when encountering an error

Reply:
There wasn't much for debugging purposes haha but your idea seems great, I like it :+1: